### PR TITLE
[quant] AddReLU module

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24217 [quant] AddReLU module**

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24217

Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

### CODE REVIEW

File: torch/nn/_intrinsic/quantized/modules/functional_relu.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import torch
+from torch.nn._intrinsic import AddReLU as NNAddReLU
+
+class AddReLU(torch.nn.Module):
+    _FLOAT_MODULE = NNAddReLU
```
Comment (jerryzh168) [inline torch/nn/_intrinsic/quantized/modules/functional_relu.py]: I think `import torch.nn._intrinsic as nni` and use `nni.AddReLU` might be more clear. I'm changing to use this pattern in my pr as well.

Also, maybe a better name for this file? functional_relu seems to be confusing.

File: torch/nn/_intrinsic/quantized/modules/functional_relu.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import torch
+from torch.nn._intrinsic import AddReLU as NNAddReLU
+
+class AddReLU(torch.nn.Module):
```
Comment (jerryzh168) [inline torch/nn/_intrinsic/quantized/modules/functional_relu.py]: Also, to reuse from_float, I think you'll need to inherit from `nnq.Add`?

**Review by jerryzh168 — CHANGES_REQUESTED** [review]
see comments

File: torch/nn/_intrinsic/quantized/modules/functional_relu.py (hunk 2 above)
Comment (zafartahirov) [reply torch/nn/_intrinsic/quantized/modules/functional_relu.py]: nnq.Add won't exist anymore -- we are depricating it in #23635

File: torch/nn/_intrinsic/quantized/modules/functional_relu.py (hunk 1 above)
Comment (zafartahirov) [reply torch/nn/_intrinsic/quantized/modules/functional_relu.py]: Suggestions for the name of the file?

Comment (zafartahirov) [reply torch/nn/_intrinsic/quantized/modules/functional_relu.py]: Resolved in person -- decided to go with the current name for now

File: torch/nn/_intrinsic/quantized/modules/__init__.py [hunk 3]
```diff
@@ -1,9 +1,12 @@
 # @lint-ignore-every PYTHON3COMPATIMPORTS
 
+from .functional_relu import AddReLU
```
Comment (jerryzh168) [inline torch/nn/_intrinsic/quantized/modules/__init__.py]: I still don't like this name, maybe just torch_relu?

File: test/common_quantization.py [hunk 4]
```diff
@@ -454,20 +455,26 @@ def __init__(self, quantized=False):
         if quantized:
             self.mycat = nnq.QFunctional()
             self.myadd = nnq.QFunctional()
+            self.my_intrinsic_add = nni.quantized.modules.AddReLU()
```
Comment (jerryzh168) [inline test/common_quantization.py]: nit: ModForWrapping -> ModelForWrapping?

File: test/common_quantization.py [hunk 5]
```diff
@@ -454,20 +455,26 @@ def __init__(self, quantized=False):
...
         z = self.myadd.add(y, y)
+        z = self.my_intrinsic_add(z, z)
         return z
 
     @classmethod
     def from_float(cls, mod):
         new_mod = cls(quantized=True)
         new_mod.mycat = new_mod.mycat.from_float(mod.mycat)
         new_mod.myadd = new_mod.myadd.from_float(mod.myadd)
+        new_mod.my_intrinsic_add = new_mod.my_intrinsic_add.from_float(mod.my_intrinsic_add)
```
Comment (jerryzh168) [inline test/common_quantization.py]: and why do we write a separate from_float rather than relying on the tool?

Comment (zafartahirov) [reply test/common_quantization.py]: I don't remember the reason :/

File: torch/nn/_intrinsic/quantized/modules/torch_relu.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import torch
+from torch.nn.quantized.modules._base_modules import _BaseQuantizedModule
+
+
+class AddReLU(torch.nn.Module, _BaseQuantizedModule):
```
Comment (jerryzh168) [inline torch/nn/_intrinsic/quantized/modules/torch_relu.py]: wait, why not a QFunctionalReLU?

Comment (jerryzh168) [reply torch/nn/_intrinsic/quantized/modules/torch_relu.py]: please create a FunctionReLU and FunctionalReLU (maybe rename them as well...)

**Review by raghuramank100 — CHANGES_REQUESTED** [review]
Please see comments

**Comment by zafartahirov** [comment]
Closing as outdated in favor of #26612

**Pull request closed without merging**